Kernel/VM/Range.cpp

```diff
@@ -55,7 +55,7 @@ Range Range::intersect(const Range& other) const
 
     // if we have no match, return an invalid range
     if (!new_size)
-        return Range(VirtualAddress(nullptr), 0ULL);
+        return {};
```

Comment:
Given that this function is now only used in a context where we expect the ranges to have non-zero intersection and would actually crash the kernel (`Optional::operator->` for `total_size_collected`), one could just move the VERIFY here, and simplify some other stuff along the way:

```
    auto new_base = max(base(), other.base());
    auto new_end = min(end(), other.end());
    VERIFY(new_base < new_end);
    return Range(new_base, new_end - new_size);
```

(And therefore no `Optional` anymore.)

Reply:
Done that